code/game/objects/structures/crates_lockers/closets/bodybag.dm

```diff
@@ -148,3 +149,159 @@
 		max_weight_of_contents = A_is_item.w_class
 	B.w_class = max_weight_of_contents
 	usr.put_in_hands(B)
+
+/// Environmental bags
+
+/obj/structure/closet/body_bag/environmental
+	name = "environmental protection bag"
+	desc = "An insulated, reinforced bag designed to protect against exoplanetary storms and other environmental factors."
+	icon = 'icons/obj/bodybag.dmi'
+	icon_state = "envirobag"
+	mob_storage_capacity = 1
+	pressure_protection = 0.8
+	thermal_insulation = 0.5
+	foldedbag_path = /obj/item/bodybag/environmental/
+	weather_protection = list(WEATHER_ACID, WEATHER_ASH, WEATHER_RAD, WEATHER_SNOW, WEATHER_VOID) // Does not protect against lava or the The Floor Is Lava spell.
+
+/obj/structure/closet/body_bag/environmental/nanotrasen
+	name = "elite environmental protection bag"
+	desc = "A heavily reinforced and insulated bag, capable of fully isolating its contents from external factors."
+	icon = 'icons/obj/bodybag.dmi'
+	icon_state = "ntenvirobag"
+	pressure_protection = 1
+	thermal_insulation = 1
+	foldedbag_path = /obj/item/bodybag/environmental/nanotrasen/
+	weather_protection = list(WEATHER_ALL)
+
+/// Securable enviro. bags
+
+/obj/structure/closet/body_bag/environmental/prisoner
+	name = "prisoner transport bag"
+	desc = "Intended for transport of prisoners through hazardous environments, this environmental protection bag comes with straps to keep an occupant secure."
+	icon = 'icons/obj/bodybag.dmi'
+	icon_state = "prisonerenvirobag"
+	foldedbag_path = /obj/item/bodybag/environmental/prisoner/
+	breakout_time = 4 MINUTES // because it's probably about as hard to get out of this as it is to get out of a straightjacket.
+	var/sinch_time = 10 SECONDS
+	var/sinched = FALSE
```

Comment:
autodoc

Reply:
elaborate?

Reply:
https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md

Reply:
read this whole file^